evap/results/tests/test_views.py

```diff
@@ -732,216 +731,199 @@ def test_private_evaluation(self):
-        self.assertNotIn(".general_additional_orig_published.", page)
-        self.assertNotIn(".general_additional_orig_hidden.", page)
-        self.assertNotIn(".general_changed_published.", page)
-        self.assertIn(".contributor_orig_published.", page)
-        self.assertNotIn(".contributor_orig_private.", page)
-        self.assertNotIn(".responsible_contributor_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_orig_hidden.", page)
-        self.assertNotIn(".responsible_contributor_orig_published_changed.", page)
-        self.assertNotIn(".responsible_contributor_changed_published.", page)
-        self.assertNotIn(".responsible_contributor_orig_private.", page)
-        self.assertNotIn(".responsible_contributor_orig_notreviewed.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_hidden.", page)
-
-    def test_textanswer_visibility_for_contributor_general_textanswers(self):
-        page = self.app.get(
-            "/results/semester/1/evaluation/1", user="contributor_general_textanswers@institution.example.com"
+        self.helper_test_contributor(user, ViewContributorResults.FULL, [])
+        self.helper_test_contributor(user, ViewContributorResults.RATINGS, [])
+        self.helper_test_contributor(user, ViewContributorResults.PERSONAL, [])
+
+    def test_responsible(self):
+
+        user = "responsible@institution.example.com"
+
+        self.helper_test_general(
+            user,
+            ViewGeneralResults.FULL,
+            [
+                ".general_orig_published.",
+                ".general_changed_published.",
+                ".general_additional_orig_published.",
+            ],
+        )
+        self.helper_test_general(user, ViewGeneralResults.RATINGS, [])
+
+        self.helper_test_contributor(user, ViewContributorResults.FULL, [])
+        self.helper_test_contributor(user, ViewContributorResults.RATINGS, [])
+        self.helper_test_contributor(user, ViewContributorResults.PERSONAL, [])
+
+    def test_responsible_contributor(self):
+
+        user = "responsible_contributor@institution.example.com"
+
+        self.helper_test_general(
+            user,
+            ViewGeneralResults.FULL,
+            [
+                ".general_orig_published.",
+                ".general_changed_published.",
+                ".general_additional_orig_published.",
+            ],
+        )
+        self.helper_test_general(user, ViewGeneralResults.RATINGS, [])
+
+        self.helper_test_contributor(
+            user,
+            ViewContributorResults.FULL,
+            [
+                ".responsible_contributor_orig_published.",
+                ".responsible_contributor_changed_published.",
+                ".responsible_contributor_orig_private.",
+                ".responsible_contributor_additional_orig_published.",
```

Comment:
For the 4 we introduced a new variable (contributor_textanswers_responsible_contributor_can_see) inside the method and use it.
We did this as well for contributor_textanswers_contributor_can_see, as this was used twice as well.
For the other 3 we introduced an instance variable (standard_general_textanswers), since this is used in multiple methods. 

Is this fine now?

Reply:
Yup, looking good, thanks! :)